### PR TITLE
Set up the Advantage page to mount a React app

### PR DESCRIPTION
**Pull request by huwshimi**
## Done

- Set up the Advantage page to mount a React app.

## QA

- Visit /advantage?test_backend=true.
- Log in.
- You should see "Your subscriptions".
- When the React app has loaded it should say "Content" under the title.

## Issue / Card

Fixes canonical-web-and-design/commercial-squad#85.

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -7,7 +7,7 @@
     "esModuleInterop": true,
     "forceConsistentCasingInFileNames": true,
     "isolatedModules": true,
-    "jsx": "react-jsx",
+    "jsx": "react",
```
Comment (huwshimi) [inline tsconfig.json:10]: It seems `esbuild` does not support the new JSX transform yet so I've reverted this.

Comment (ClementChaumel) [reply tsconfig.json:10]: What was the issue?

**Review by ClementChaumel — APPROVED** [review]
:+1:

File: tsconfig.json (hunk 1 above)
Comment (huwshimi) [reply tsconfig.json:10]: It seems that esbuild has to add support for the React 17 JSX transform so I guess it just wasn't going to build with `"react-jsx"` (there were conflicting errors about needing React to be imported/not imported).

More details in the issue: https://github.com/evanw/esbuild/issues/334.

**Pull request merged**